applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_operations.cpp

```diff
@@ -0,0 +1,166 @@
+    
+    // Define the nodal values
+    Vector potential(3);
+    potential(0) = 1.0;
+    potential(1) = 2.0;
+    potential(2) = 3.0;
+    for (unsigned int i = 0; i < 3; i++){
+      if (wake_elemental_distances(i) > 0.0)
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i);
+      else
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL) = potential(i);
+    }
+    for (unsigned int i = 0; i < 3; i++){
+      if (wake_elemental_distances(i) < 0.0)
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i)+1;
+      else
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL) = potential(i)+1;
+    }      
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    
+    // Create fake compressible_model_part
+    ModelPart& compressible_model_part = this_model.CreateModelPart("Compressible", 3);
+    compressible_model_part.GetProcessInfo()[DOMAIN_SIZE] = 2;
+    
+    // Variables addition
+    compressible_model_part.AddNodalSolutionStepVariable(DENSITY);
+    compressible_model_part.AddNodalSolutionStepVariable(MOMENTUM);
+    compressible_model_part.AddNodalSolutionStepVariable(TOTAL_ENERGY);
+    
+    // Set the element properties
+    compressible_model_part.CreateNewProperties(0);
+    Properties::Pointer compressible_pElemProp = compressible_model_part.pGetProperties(0);
+    
+    // Geometry creation
+    compressible_model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    compressible_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    compressible_model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+    std::vector<ModelPart::IndexType> elemNodes_2{ 1, 2, 3 };
+    compressible_model_part.CreateNewElement("IncompressiblePotentialFlowElement2D3N", 1, elemNodes_2, compressible_pElemProp);
+    
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    
+    // Operation parameters
+    Parameters operation_parameters = Parameters(R"(
+      {
+      "origin_model_part"       : "Main",
+      "destination_model_part"  : "Compressible",
+      "reference_temperature"   : 273,
+      "compute_nodal_velocities": true
+      })" );
+    
+    // Execute PotentialToCompressibleNavierStokesOperation
+    PotentialToCompressibleNavierStokesOperation PotentialToCompressibleNavierStokesOperation(this_model, operation_parameters);
+    PotentialToCompressibleNavierStokesOperation.Execute();
+    
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+   
+    for(unsigned int index = 0; index < model_part.NumberOfNodes(); ++index)
+    {  
+      auto r_node = compressible_model_part.NodesBegin() + index;
+      const auto potential_node = model_part.NodesBegin() + index;
```

Comment:
```suggestion
      auto it_node = compressible_model_part.NodesBegin() + index;
      const auto it_potential_node = model_part.NodesBegin() + index;
```
These are iterators.